### PR TITLE
Reggieg

### PR DESCRIPTION
**Pull request by regedit32**
Updates to NTFSAccessEntry resource (Added Get, Set, Test)

### CODE REVIEW

File: DscResources/NTFSAccessEntry/NTFSAccessEntry.psm1 [hunk 1]
```diff
@@ -0,0 +1,514 @@
...
+        {
+            $currentAcl.RemoveAccessRule($Rule)
+        }
+        catch
+        {
+            try
+            {
+                #If failure due to Idenitty translation issue then create the same rule with the identity as a sid to remove account
+                $SID = ConvertTo-SID -IdentityReference $Rule.IdentityReference.Value
+                $SIDRule = [System.Security.AccessControl.RegistryAccessRule]::new($SID, $Rule.RegistryRights.value__, $Rule.InheritanceFlags.value__, $Rule.PropagationFlags.value__, $Rule.AccessControlType.value__)
```
Comment (mcollera) [inline DscResources/NTFSAccessEntry/NTFSAccessEntry.psm1]: This has to be changes to filesystem

File: DscResources/NTFSAccessEntry/NTFSAccessEntry.psm1 [hunk 2]
```diff
@@ -0,0 +1,514 @@
...
+    $results = @()
+    $ToBeRemoved = @()
+    $AbsentToBeRemoved = @()
+
+    $PresentRules = $Expected.Where({$_.Ensure -eq 'Present'}).Rules
+    $AbsentRules = $Expected.Where({$_.Ensure -eq 'Absent'}).Rules
+    foreach($refrenceObject in $PresentRules)
+    {
+        $match = $Actual.Where({
+            $_.RegistryRights -eq $refrenceObject.RegistryRights -and
```
Comment (mcollera) [inline DscResources/NTFSAccessEntry/NTFSAccessEntry.psm1]: Needs to be updated for filesystem rights

File: DscResources/NTFSAccessEntry/NTFSAccessEntry.psm1 [hunk 3]
```diff
@@ -0,0 +1,514 @@
...
+                Rule = $refrenceObject
+                Match = $false
+            }
+        }
+    }
+
+    foreach($refrenceObject in $Actual)
+    {
+        $match = $Expected.Rules.Where({
+            $_.RegistryRights -eq $refrenceObject.RegistryRights -and
```
Comment (mcollera) [inline DscResources/NTFSAccessEntry/NTFSAccessEntry.psm1]: Same

File: DscResources/NTFSAccessEntry/NTFSAccessEntry.psm1 [hunk 4]
```diff
@@ -0,0 +1,514 @@
...
+            $ToBeRemoved += [PSCustomObject]@{
+                Rule = $refrenceObject
+            }
+        }
+    }
+
+    foreach($refrenceObject in $AbsentRules)
+    {
+        $match = $Actual.Where({
+            $_.RegistryRights -eq $refrenceObject.RegistryRights -and
```
Comment (mcollera) [inline DscResources/NTFSAccessEntry/NTFSAccessEntry.psm1]: Same

File: DscResources/RegistryAccessEntry/RegistryAccessEntry.psm1 [hunk 5]
```diff
@@ -0,0 +1,296 @@
+Import-Module -Name (Join-Path -Path ( Split-Path $PSScriptRoot -Parent ) `
```
Comment (mcollera) [inline DscResources/RegistryAccessEntry/RegistryAccessEntry.psm1]: Should remove from your pull request

File: Examples/Sample_AccessControlEntry_NTFS.ps1 [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+                            Ensure = 'Present'
+                        }
+                    )               
+                }
+            )
+        }
+    }
+}
+
+$session = New-PSSession -ComputerName $TargetName -Credential $credential
```
Comment (mcollera) [inline Examples/Sample_AccessControlEntry_NTFS.ps1]: We will need to remove this stuff when we publish

**Comment by regedit32** [comment]
closing this pull request and creating new one that excludes regsitryaccess files

**Pull request closed without merging**